flink-clients/src/test/java/org/apache/flink/client/ExecutionEnvironmentTest.java

```diff
@@ -91,9 +106,42 @@ public void testUserDefinedJobNameWithConfigure() {
         testJobName(jobName, env);
     }
 
+    @Test
+    public void testConfigureFileSystem() throws Exception {
+        final Configuration config = new Configuration();
+        EXPECTED_CONFIGURATION.forEach(config::setString);
+        final StreamExecutionEnvironment environment = new LocalStreamEnvironment(config);
+        environment.getCheckpointConfig().setCheckpointStorage("testConfig:/" + tmp.newFolder());
+        environment.enableCheckpointing(100);
+        environment.fromElements(1, 2, 3, 4).map((MapFunction<Integer, Integer>) value -> value);
+        // We only want to enable the configuration check for this test case. Nevertheless, the
+        // filesystem is loaded and configured for all test cases in this module.
+        expectConfiguration = true;
+        environment.execute("configureFilesystem");
+        expectConfiguration = false;
+    }
+
     private void testJobName(String prefixOfExpectedJobName, ExecutionEnvironment env) {
         env.fromElements(1, 2, 3).writeAsText("/dev/null");
         Plan plan = env.createProgramPlan();
         assertTrue(plan.getJobName().startsWith(prefixOfExpectedJobName));
     }
+
+    /** Test filesystem to check that configurations are propagated. */
+    public static final class TestFileSystemFactoryWithConfiguration
+            extends TestFileSystem.TestFileSystemFactory {
+
+        @Override
+        public String getScheme() {
+            return "testConfig";
+        }
+
+        @Override
+        public void configure(Configuration config) {
+            if (!expectConfiguration) {
+                return;
+            }
+            EXPECTED_CONFIGURATION.forEach((k, v) -> assertEquals(v, config.getString(k, null)));
```

Comment:
Use assertion string.

Reply:
Sorry but what exactly do you want to see here? :)

Reply:
```
assertEquals(v, config.getString(k, null), "Unexpected config entry for " + k)
```